### PR TITLE
upgrade denendencies

### PR DESCRIPTION
**Pull request by niedhui**
## What have you changed? (mandatory)
upgrade following dependencies:
  regex(0.1 to 1.2)
  chrono(0.4 to 0.4.5)
  clap(2 to 2.32.0)
  signal(0.4 to 0.6.0)
  serde_json(1.0 to 1.0.26)
  byteorder(0.5 to 1.2.4)
  crc(1.2 to 1.8.1)
  nix(0.9.0 to 0.11.0)
  crossbeam-channel(0.2 to 0.2.4)

I upgrade some denendencies to reduce dependency version conflicts, due to  #3146 
there still some conflicts, blocked by `mio`(will be removed when threaded raftstore is landed), 
`prometheus`, `raft`, and `fail`(these three need to upgrade dependeneis first).

## What are the type of the changes? (mandatory)
Improvement

## How has this PR been tested? (mandatory)
unit test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

### CODE REVIEW

**Comment by siddontang** [comment]
Please fix the conflict.

**Comment by breeswish** [comment]
Thanks a lot! For prometheus, you can send us PRs as well! https://github.com/pingcap/rust-prometheus  Also please resolve conflicts.

**Comment by niedhui** [comment]
conflicts fixed

File: Cargo.toml [hunk 1]
```diff
@@ -37,25 +37,25 @@ slog-async = "2.3"
 slog-scope = "4.0"
 slog-stdlog = "3.0"
 slog-term = "2.4"
-byteorder = "0.5"
+byteorder = "1.2.4"
```
Comment (hicqu) [inline Cargo.toml]: Why specify the minor version? I think `1.2` is enough.

Comment (niedhui) [reply Cargo.toml]: I use `cargo upgrade <dep>` to upgrade the crate, it simply add the newest version(include minor verions) automatically, just like `cargo add` or copy from the crates.io .  I checked there is some crate also specify the minor version before, like `lazy_static = "0.2.1"` , most others without minor version.
If there is a convention, please let me know, I will remove the minor version. thanks

Comment (breeswish) [reply Cargo.toml]: Thanks! I agree with @hicqu , generally we should avoid using minor versions.

File: Cargo.lock [hunk 2]
```diff
@@ -1877,11 +1870,10 @@ dependencies = [
 "checksum miow 0.2.1 (registry+https://github.com/rust-lang/crates.io-index)" = "8c1f2f3b1cf331de6896aabf6e9d55dca90356cc9960cca7eaaf408a355ae919"
 "checksum murmur3 0.4.0 (git+https://github.com/pingcap/murmur3.git)" = "<none>"
 "checksum net2 0.2.23 (registry+https://github.com/rust-lang/crates.io-index)" = "6a816012ca11cb47009693c1e0c6130e26d39e4d97ee2a13c50e868ec83e3204"
+"checksum nix 0.11.0 (registry+https://github.com/rust-lang/crates.io-index)" = "d37e713a259ff641624b6cb20e3b12b2952313ba36b6823c0f16e6cfd9e5de17"
```
Comment (breeswish) [inline Cargo.lock]: Seems that old nix still exists

**Comment by breeswish** [comment]
@hicqu PTAL

**Pull request merged**